transformations/whitespace_perturbation/transformation.py

```diff
@@ -0,0 +1,33 @@
+import random
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+def whitespace(text, remove_prob=0.1, add_prob=0.05, seed=0):
+    random.seed(seed)
+    newtext = ''
```

Comment:
You should use a list to accumulate chars and join at the end to avoid creating a new string at every step.

Reply:
Good catch! I updated the code per your suggestion.